src/main/java/net/sf/jabref/logic/citationstyle/CitationStyleGenerator.java

```diff
@@ -0,0 +1,71 @@
+package net.sf.jabref.logic.citationstyle;
+
+import java.io.IOException;
+import java.util.Map;
+
+import net.sf.jabref.logic.formatter.bibtexfields.UnicodeToLatexFormatter;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.model.entry.BibEntry;
+
+import de.undercouch.citeproc.CSL;
+import de.undercouch.citeproc.bibtex.BibTeXConverter;
+import de.undercouch.citeproc.csl.CSLItemData;
+import de.undercouch.citeproc.output.Bibliography;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.jbibtex.BibTeXEntry;
+import org.jbibtex.DigitStringValue;
+import org.jbibtex.Key;
+import org.jbibtex.TokenMgrError;
+
+
+/**
+ * WARNING: the citation is generated with JavaScript which may take some time, better call it in outside the main Thread
+ */
+public class CitationStyleGenerator {
+
+    private static final Log LOGGER = LogFactory.getLog(CitationStyleGenerator.class);
+    private static final UnicodeToLatexFormatter UNICODE_TO_LATEX_FORMATTER = new UnicodeToLatexFormatter();
+    private static final BibTeXConverter BIBTEX_CONVERTER = new BibTeXConverter();
+
+
+    /**
+     * Generates a Citation based on the given entry and style
+     * WARNING: the citation is generated with JavaScript which may take some time, better call it in outside the main Thread
+     */
+    static String generateCitation(BibEntry entry, String style) {
+        return generateCitation(entry, style, CitationStyleOutputFormat.HTML);
+    }
+
+    /**
+     * Generates a Citation based on the given entry, style, and output format
+     * WARNING: the citation is generated with JavaScript which may take some time, better call it in outside the main Thread
+     */
+    static String generateCitation(BibEntry entry, String style, CitationStyleOutputFormat outputFormat) {
+        try {
+            BibTeXEntry bibTeXEntry = new BibTeXEntry(new Key(entry.getType()), new Key(entry.getCiteKey()));
+            for (Map.Entry<String, String> field : entry.getFieldMap().entrySet()) {
+                String value = UNICODE_TO_LATEX_FORMATTER.format(field.getValue());
+                bibTeXEntry.addField(new Key(field.getKey()), new DigitStringValue(value));
+            }
+
+            CSLItemData cslItemData = BIBTEX_CONVERTER.toItemData(bibTeXEntry);
+            Bibliography bibliography = CSL.makeAdhocBibliography(style, outputFormat.getFormat(), cslItemData);
+            return bibliography.getEntries()[0];
+
+        } catch (IOException  e) {
+            LOGGER.error("Could not generate BibEntry Citation", e);
+        } catch (TokenMgrError e) {
+            LOGGER.error("Bad character inside BibEntry", e);
+            // sadly one can not easily retrieve the bad char from the TokenMgrError
+            return  new StringBuilder()
+                    .append(Localization.lang("Bad character inside BibEntry"))
```

Comment:
Just look at other localizations.

[Please write cannot](https://en.oxforddictionaries.com/usage/cannot-or-can-not)